### PR TITLE
PO stat button: show all the related shipment

### PR DESCRIPTION
**Pull request by jpr-odoo**
Description of the issue/feature this PR addresses:
Task: https://www.odoo.com/web?#id=31061&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.23282abb4ded8664dfce189d83cebe11
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/views/purchase_views.xml [hunk 1]
```diff
@@ -160,7 +160,7 @@
                     <button name="button_confirm" type="object" states="sent" string="Confirm Order" class="oe_highlight" id="bid_confirm"/>
                     <button name="button_approve" type="object" states='to approve' string="Approve Order" class="oe_highlight" groups="purchase.group_purchase_manager"/>
                     <button name="action_rfq_send" states="purchase" string="Send PO by Email" type="object" context="{'send_rfq':False}"/>
-                    <button name="action_view_picking" string="Receive Products" class="oe_highlight" type="object" attrs="{'invisible': ['|', '|' , ('is_shipped', '=', True), ('state','not in', ('purchase','done')), ('picking_count', '=', 0)]}"/>
+                    <button name="action_view_incoming_picking" string="Receive Products" class="oe_highlight" type="object" attrs="{'invisible': ['|', '|' , ('is_shipped', '=', True), ('state','not in', ('purchase','done')), ('picking_count', '=', 0)]}"/>
```
Comment (sle-odoo) [inline addons/purchase/views/purchase_views.xml]: is action_view_picking still used? i see it's used in the stat button

File: addons/purchase/tests/test_purchase_order.py [hunk 2]
```diff
@@ -156,4 +155,4 @@ def test_02_po_return(self):
         self.invoice.invoice_line_ids[0].quantity = 2.0
         self.invoice.invoice_line_ids[1].quantity = 2.0
         self.invoice.invoice_validate()
-        self.assertEqual(self.po.order_line.mapped('qty_invoiced'), [3.0, 3.0], 'Purchase: Billed quantity should be 3.0')
\ No newline at end of file
+        self.assertEqual(self.po.order_line.mapped('qty_invoiced'), [3.0, 3.0], 'Purchase: Billed quantity should be 3.0')
```
Comment (sle-odoo) [inline addons/purchase/tests/test_purchase_order.py]: useless diff in this file

File: addons/purchase/models/purchase.py [hunk 3]
```diff
@@ -86,13 +86,18 @@ def _compute_picking(self):
                 # do some recursive search, but that could be prohibitive if not done correctly.
                 moves = line.move_ids | line.move_ids.mapped('returned_move_ids')
                 pickings |= moves.mapped('picking_id')
+            # show all shipment related to current purchase order
+            if order.group_id:
+                pickings |= self.env['stock.picking'].search([('group_id', '=', order.group_id.id)])
```
Comment (sle-odoo) [inline addons/purchase/models/purchase.py]: in line 87 and 88 should we not only look for returned move? i guess this search will also gives back the first receipt

maybe you can add picking_ids o2m on procurement group to be cleaner?

**Comment by hmo-odoo** [comment]
cancelled

**Pull request closed without merging**